### DISCUSSION TITLE
evolve to IPvN

### DISCUSSION THREAD

**Discussion by xialvjun** [yggdrasil-network/yggdrasil-go#1053] [Ideas]
IPv6 has just 128 bits, and yggdrasil-network has only 64 bits for the public key, that makes people concerned.

How about we build yggdrasil into system core as IPvN protocol.

The IP address is totally it's public key(`algorithm version + public key`), and even it's length is dynamic (we shouldn't worry about the ip it too long may make network ineffecient because it's an overlay network on the real network, the MTU is unlimited).

**Comment by majestrate** [comment]
if there was an IETF RFC for variable sized network addresses I think that'd be a place to start researching platform support of an IPvN. the reason everyone uses IPv6 in these experimental protocols is because of the network effect (popularity) of support in platforms.

**Reply by xialvjun** [reply]
But in the foreseeable future, few projects other than the Yggdrasil will need IPvN.

Well, it seems that even yggdrasil has add IPvN to system core, the application layer still can not use it unless the app support it in code.
